app/assets/javascripts/discourse/app/models/user.js

```diff
@@ -1072,6 +1072,10 @@ User.reopenClass(Singleton, {
     return ajax(userPath("check_email"), { data: { email } });
   },
 
+  resetRecentSearches() {
+    return ajax(`/u/reset-recent-searches`, { type: "POST" });
```

Comment:
Minor but maybe it would be more RESTful to say `DELETE` at ` /u/recent-searches`?